hack/deployer/runner/gcloud.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runner
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	GCPDir                 = ".gcp"
+	ServiceAccountFilename = "osServiceAccount.json"
+)
+
+// authToGCP authenticates the deployer to the Google Cloud Platform as a service account or as a user.
+func authToGCP(
+	vaultInfo *VaultInfo, vaultPath string, serviceAccountVaultFieldName string,
+	asServiceAccount bool, useNonDefaultCloudSDKPath bool, configureDocker bool, gCloudProject interface{},
+) error {
+	if vaultInfo == nil {
```

Comment:
I think `vaultInfo` is required only if authenticating as a service account _(it is a least true for GKE)_. Could we do this check only when `if asServiceAccount` is `true` ?

Reply:
Sure! I went a bit fast on this one.